# Don't emit notify::volume twice when calling gvc_mixer_stream_set_volume (). The second
# emission triggers the osd-notification before the channel is {un}muted with
# gvc_mixer_stream_change_is_muted () when the volume is changed to/from zero.
# This causes a confusing out-of-sync notification with a mute icon and a volume of greater
# than zero.

diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.27.90/plugins/media-keys/cut-n-paste/gvc-mixer-stream.c gnome-settings-daemon-2.27.90.new/plugins/media-keys/cut-n-paste/gvc-mixer-stream.c
--- gnome-settings-daemon-2.27.90/plugins/media-keys/cut-n-paste/gvc-mixer-stream.c	2009-08-12 10:15:00.000000000 +0100
+++ gnome-settings-daemon-2.27.90.new/plugins/media-keys/cut-n-paste/gvc-mixer-stream.c	2009-08-14 15:22:25.032298400 +0100
@@ -154,7 +154,6 @@
 
         if (!pa_cvolume_equal(gvc_channel_map_get_cvolume(stream->priv->channel_map), &cv)) {
                 gvc_channel_map_volume_changed(stream->priv->channel_map, &cv);
-                g_object_notify (G_OBJECT (stream), "volume");
         }
 
         return TRUE;
